sdks/python/apache_beam/runners/dataflow/internal/apiclient.py

```diff
@@ -177,6 +177,14 @@ def __init__(self, packages, options, environment_version, pipeline_url):
     if self.debug_options.experiments:
       for experiment in self.debug_options.experiments:
         self.proto.experiments.append(experiment)
+    # Add use_multiple_sdk_containers flag if its not already present. Do not
+    # add the flag if 'no_use_multiple_sdk_containers' is present.
+    # TODO: Cleanup use_multiple_sdk_containers once we deprecate Python SDK
+    # till version 2.4.
+    if (job_type.startswith('FNAPI_') and
+        'use_multiple_sdk_containers' not in self.proto.experiments and
+        'no_use_multiple_sdk_containers' not in self.proto.experiments):
+      self.proto.experiments.append('use_multiple_sdk_containers')
```

Comment:
It is preferable to modify debug_options.experiments (as done above for `runner_harness_override`). This also properly helps with updating the user visible pipeline options in the UI, and it will auto added to the proto by the loop above.

It would also help combine things related to `if job_type.startswith('FNAPI_'):` in a single place.